kratos/geometries/quadrature_point_surface_in_volume_geometry.h

```diff
@@ -0,0 +1,277 @@
+
+    typedef QuadraturePointGeometry<TPointType, 3, 3, 2> BaseType;
+    typedef Geometry<TPointType> GeometryType;
+
+    typedef typename GeometryType::IndexType IndexType;
+    typedef typename GeometryType::SizeType SizeType;
+
+    typedef typename GeometryType::PointsArrayType PointsArrayType;
+    typedef typename GeometryType::CoordinatesArrayType CoordinatesArrayType;
+
+    typedef GeometryShapeFunctionContainer<GeometryData::IntegrationMethod> GeometryShapeFunctionContainerType;
+    typedef BoundedMatrix<double,3,2> TangentMatrixType;
+
+    /// using base class functions
+    using BaseType::Jacobian;
+    using BaseType::DeterminantOfJacobian;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Constructor with points and geometry shape function container
+    QuadraturePointSurfaceInVolumeGeometry(
+        const PointsArrayType& ThisPoints,
+        GeometryShapeFunctionContainerType& ThisGeometryShapeFunctionContainer,
+        TangentMatrixType LocalTangents )
+        : BaseType(ThisPoints, ThisGeometryShapeFunctionContainer)
+        , mLocalTangents(LocalTangents)
+    {
+    }
+
+    /// Constructor with points, geometry shape function container and parent
+    QuadraturePointSurfaceInVolumeGeometry(
+        const PointsArrayType& ThisPoints,
+        GeometryShapeFunctionContainerType& ThisGeometryShapeFunctionContainer,
+        TangentMatrixType LocalTangents,
+        GeometryType* pGeometryParent)
+        : BaseType(ThisPoints, ThisGeometryShapeFunctionContainer, pGeometryParent)
+        , mLocalTangents(LocalTangents)
+    {
+    }
+
+    /// Destructor.
+    ~QuadraturePointSurfaceInVolumeGeometry() override = default;
+
+    /// Copy constructor.
+    QuadraturePointSurfaceInVolumeGeometry(QuadraturePointSurfaceInVolumeGeometry const& rOther )
+        : BaseType( rOther )
+        , mLocalTangents(rOther.mLocalTangents)
+    {
+    }
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    /// Assignment operator.
+    QuadraturePointSurfaceInVolumeGeometry& operator=( const QuadraturePointSurfaceInVolumeGeometry& rOther )
+    {
+        BaseType::operator=( rOther );
+
+        mLocalTangents = rOther.mLocalTangents;
+
+        return *this;
+    }
+
+    ///@}
+    ///@name Dynamic access to internals
+    ///@{
+
+    /// Assign with Matrix
+    void Assign(
+        const Variable<Matrix >& rVariable,
+        const Matrix& rInput) override
+    {
+        KRATOS_ERROR_IF( rInput.size1() != mLocalTangents.size1() || rInput.size2() != mLocalTangents.size2() )
+            << "QuadraturePointSurfaceInVolume :: Input Matrix does not have the same size as LocalTangentMatrix. Size must be (3,2)." << std::endl;
+
+        if (rVariable == LOCAL_TANGENT_MATRIX) {
+            mLocalTangents = rInput;
+        }
+    }
+
+    /// Calculate with Matrix
+    void Calculate(
+        const Variable<Matrix >& rVariable,
+        Matrix& rOutput)
```

Comment:
not sure why the CI complains but you must have the `const override` here.

Reply:
Yes, this was just a try. I thought it is related to this.

Reply:
Fixed :punch: